### ISSUE TITLE
Firebase functions causes Gmail to block login attempt

### ISSUE THREAD

**Issue by Isoaxe** [nodemailer/nodemailer#1273]
Please fill the following questionnaire about your issue:

1. What kind of issue are you reporting?
- [ ] A bug in Nodemailer
- [ ] A bug in a plugin of Nodemailer (eg. issues with nodemailer-sendgrid)
- [ ] Feature request
- [X] Looking for help to resolve some kind of problem with Nodemailer

2. Are you listed as a sponsor of Nodemailer project (see Sponsors button above)?
- [ ] Yes. Sponsors get priority support
- [X] No. Unless it is a bug in Nodemailer you might find support from public forums like [StackOverflow](https://stackoverflow.com/search?q=nodemailer).

3. State your problem here:

This concerns a simple contact form that involves sending two emails on reaching the `/submit` endpoint with a `POST` request. There are no issues when hosted locally or even hosted remote via an Express server. However, when hosting on Firebase and utilising Firebase functions, the login attempt by Gmail is blocked and results in the following error: `Process exited with code 16 at process`.

The issue is outlined more fully [here](https://stackoverflow.com/questions/66693367/firebase-functions-with-nodemailer-post-request-error-process-exited-with-code). Perhaps there is a workaround for this? I've tried using other service providers such as Yahoo and Outlook365 but run into additional issues there too.

**Comment by andris9** [comment]
Most probably what happens there is that Gmail rejects your authentication due to bot detection heuristics and you have no error handling for the `sendMail` method which results in an uncaught error.

**Comment by Isoaxe** [comment]
@andris9 Thanks for the response. I thought I would follow up in case others are encountering a similar issue as this has now been resolved.

I needed to both enable less secure apps and display unlock captcha on the gmail account, only the former of which I had done. I also needed to set the `gmail.email` and `gmail.password` Google cloud environment variables.

A great resource I found on this is an [official Firebase function sample](https://github.com/firebase/functions-samples/tree/master/quickstarts/email-users) that utilises Nodemailer.
